add seo tags to activist portal

Vector-Hector opened the pull request:
## Description
This PR adds relevant search engine optimization, so the pages get displayed right. These are also relevent for sharing events and stuff on social media. 

## Changes
[Add a list of features added/changed, bugs fixed etc]

* Adds `seoTags.ts` that includes most common tags
* Changes `layout.tsx` in these pages: My Zetkin, event, organization, projects, survey
* Adds some descriptions to messages
* Changes an import at `messageIds.ts`. See https://github.com/zetkin/app.zetkin.org/pull/3164 for why

## Notes to reviewer
[Add instructions for testing]
If you want to see what it actually changes, for the titles, you can see them in the browser tab. But for everything else, you'd need to look into the rendered html. Or deploy it to some staging server and link the pages on social media.

richardolsson commented on src/utils/seoTags.ts:
```diff
@@ -0,0 +1,95 @@
+    'equity',
+    'campaigns',
+    'events',
+  ],
```
I appreciate the eagerness to improve SEO. We're starting from scratch, so any addition is an improvement. But we also have to weigh that against other important perspectives.

We don't usually (or ever?) put content or meta-data like this on the pages that our partner organizations create. UI labels sure, but all _content_ in Zetkin, especially all that is visible on the activist portal pages, is user-generated.

I think that makes sense, because we have a large number of partner organizations, which would not be able to agree on keywords if we asked them.

> Telling it the page is about microphones when it is actually a tree planting action would be bad, but telling it it's something about sustainable initiatives is good.

Who gets to decide which keywords fall within scope of what the page is about, and which fall outside? Keywords that are obvious to some of our partner organizations would be controversial for another. Picking words to describe an organization is very much political.

Can you see why I find it problematic for us to arbitrarily pick some keywords for the content of our partner organizations?

richardolsson replied:
> And I actually wonder how you intend people to get to the feed if it's not on their search engine? Would they need to search for an organization, click their user avatar and then go to the feed?

I believe I understand where this question is coming from. I know you've raised multiple times that the activist portal is not very easy to navigate spontaneously. You're right about this, and the reason is that because it's not the primary activist portal yet. The Gen3 web app has only a few entry points (where you have to know the URL) and several exit points leading back to the old Gen2.

The old, Gen2 activist portal is still the one that most people use, and people are only finding app.zetkin.org (or the equivalent on other instances) when the instance owner actively informs them about it.

This is beginning to change. We are wrapping up the final pieces of the Gen3 activist portal (account management stuff). And even sooner than that we will start redirecting users to the /my pages (where the feed is) immediately when they log in, instead of /organize.

So finding the "my" pages will very soon by much easier, as a next step towards making the Gen3 activist portal the default (but multiple steps remain).

I truly do not understand why we would make the user "my pages" search engine optimized, or even indexable at all. I have tried googling it and can't find anything that clearly suggests that's the best thing to do. But when I read your comments, I perceive you as very confident about it, so I'm wondering if that's because I'm missing something or if it's just a matter of different communication styles.

If you still feel strongly that the authenticated-only, personalized "my" pages should be indexed, can you please elaborate on why?

> I actually was curious and looked what YouTube looks like for search engines and it's just the skeleton loading indicator. The content is loaded afterwards by js. The description and tag language depends on your IP and the Accept-Language header.

I believe this is different, but still very relevant. The youtube "feed" URL is also the root URL and the public home page. Zetkin doesn't have a page like that.

We are however right now considering what the "home" page of Zetkin should be like, if indeed there should be one at all. As part of that process, we need to think about SEO, which is something that we have not considered so far. What you've pointed out here highlights that.

Vector-Hector replied:
> But when I read your comments, I perceive you as very confident about it, so I'm wondering if that's because I'm missing something or if it's just a matter of different communication styles.

Probably the communication styles. I did also notice we have different communication styles. I'm very literal and clear. I know that can throw people off. I didn't say my preference in the first comment https://github.com/zetkin/app.zetkin.org/pull/3165#discussion_r2500575195 because I was kind of aware of that I don't know how that redirect restructure will work exactly. But if you then ask me
> Do you think we should render those pages to search engines

I read that as you want me to state what my choice would be based on what I know. And therefore, I stated what that choice would be and why.

Well anyway, I think it is a good sign. Communication styles align over time I think and differences indicate differences in thinking styles. And diversity in thinking styles makes great products.

I think my opinion on this does depend heavily on how the landing page is gonna look and how obvious it is to get to the feed from there. If it's hard or non-obvious, index it on search engines. Otherwise, don't. Currently there is no landing page, so my choice would be index it. Though I guess it doesn't make sense if you expect it to change soon. 

So basically, I am on board with not indexing them.

Vector-Hector commented:
Alrighty, I made tags optional and also removed them from surveys. The seo description in the organization page had the same problem as the tags, so I removed that as well. Something users probably should be able to configure as well down the line.

Regarding the /my pages, I set index: false, which tells search engines to not index this page. Since the description tag is then also unnecessary, I removed that as well. I kept my changes to the title though, because I think it reads better than just "My Zetkin", especially in the browser history when switching between tabs.

## Updated changes
[Add a list of features added/changed, bugs fixed etc]

* Adds `seoTags.ts` that includes most common tags
* Changes `layout.tsx` in these pages: My Zetkin, event, organization, projects, survey
